python/cuml/thirdparty_adapters/sparsefuncs_fast.py

```diff
@@ -0,0 +1,340 @@
+#
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+import cupy as cp
+from numba import cuda
+from math import ceil
+
+
+def csr_mean_variance_axis0(X):
+    """Compute mean and variance
```

Comment:
across axis 0 only presumably?

Also note the need for blank lines in comments in this file before sections